### PR TITLE
feat: move some flags into configs for provider and deploy

### PR DESCRIPTION
**Pull request by shamsartem**

### CODE REVIEW

File: docs/commands/README.md [hunk 1]
```diff
@@ -231,19 +231,17 @@ Deploy workers according to deal in 'deals' property in fluence.yaml
 
 ```
 USAGE
-  $ fluence deal deploy [WORKER-NAMES] --collateral-per-worker <value> --min-workers <value> --target-workers
-    <value> --max-workers-per-provider <value> --price-per-worker-epoch <value> [--no-input] [-k <value>]
-    [--off-aqua-logs] [--priv-key <value>] [--env <value>] [--relay <value>] [--ttl <value>] [--dial-timeout <value>]
-    [--particle-id] [--import <value>] [--no-build] [--tracing] [--marine-build-args <value>] [--auto-match]
```
Comment (akim-bow) [inline docs/commands/README.md:237]: kaef

File: src/commands/deal/create.ts [hunk 2]
```diff
@@ -66,11 +66,11 @@ export default class Create extends BaseCommand<typeof Create> {
 
     const dealAddress = await dealCreate({
       appCID: flags["app-cid"],
-      collateralPerWorker: flags["collateral-per-worker"],
+      collateralPerWorker: Number(flags["collateral-per-worker"]),
       minWorkers: flags["min-workers"],
       targetWorkers: flags["target-workers"],
       maxWorkersPerProvider: flags["max-workers-per-provider"],
-      pricePerWorkerEpoch: flags["price-per-worker-epoch"],
+      pricePerWorkerEpoch: Number(flags["price-per-worker-epoch"]),
```
Comment (akim-bow) [inline src/commands/deal/create.ts:73]: Do you validate it somehow? What if string contains float value for example?

File: src/commands/deal/deploy.ts [hunk 3]
```diff
@@ -162,8 +142,19 @@ export default class Deploy extends BaseCommand<typeof Deploy> {
       definition: appCID,
     } of uploadResult.workers) {
       const deal = fluenceConfig.deals?.[workerName];
-      assert(deal !== undefined);
-      const { minWorkers = MIN_WORKERS, targetWorkers = TARGET_WORKERS } = deal;
+
+      assert(
+        deal !== undefined,
+        "Unreachable. All worker names are checked above",
+      );
```
Comment (akim-bow) [inline src/commands/deal/deploy.ts:149]: I don't like this assert. If you made validation already, probably your validating function should return refined `fluenceConfig.deals` like a proof of validation? Probably every validation should either throw or return refined value e. g. without union with `undefined` or `null`.

File: src/lib/configs/project/provider.ts [hunk 4]
```diff
@@ -431,11 +449,19 @@ export async function initNewReadonlyProviderConfig({
...
+export function initProviderConfig(name?: string | undefined) {
+  if (name !== undefined) {
+    setProviderConfigName(name);
+  }
+
   return getConfigInitFunction(getInitConfigOptions())();
 }
 
-export function initReadonlyProviderConfig() {
+export function initReadonlyProviderConfig(name?: string | undefined) {
```
Comment (akim-bow) [inline src/lib/configs/project/provider.ts:460]:
```suggestion
export function initReadonlyProviderConfig(name?: string) {
```

Looks like in js omitting argument is the same as passing `undefined` directly

File: src/commands/deal/create.ts (hunk 2 above)
Comment (shamsartem) [reply src/commands/deal/create.ts:73]: this command is hidden from users and only for development purposes, so I don't validate anything here

File: src/lib/configs/project/provider.ts (hunk 4 above)
Comment (shamsartem) [reply src/lib/configs/project/provider.ts:460]: Yeah for function params this seems to be true, will apply this imporvement. This is not true for objects though, cause for objects a new key might be created with undefined inside, which would a different runtime behaviour

**Comment by akim-bow** [comment]
Looks nice!

**Pull request merged**